extension/visual.lua

```diff
@@ -1,89 +1,196 @@
+			points[iteration + 1] = TransformToParentPoint(transform, Vec(sin(((v + 360 / sides) + rotation) * degreeToRadian) * radius, 0, cos(((v + 360 / sides) + rotation) * degreeToRadian) * radius))
+			if iteration > 2 then
+				DrawFunction(points[iteration], points[iteration + 1], r, g, b, a)
+			end
+			iteration = iteration + 2
+		end
+		
+		return points
+    end
+
+	function visual.drawprism(transform, quat, radius, depth, rotation, sides, info)
+		local points = {}
+		local iteration = 1
+		local sqrt, sin, cos, max = math.sqrt, math.sin, math.cos, math.max
+		local r, g, b, a
+		local DrawFunction = DrawLine
+
+		if not transform.pos then transform = Transform(transform, quat or QUAT_ZERO) end
+
+		if info then
+			r = info.r and info.r or 1
+			g = info.g and info.g or 1
+			b = info.b and info.b or 1
+			a = info.a and info.a or 1
+			DrawFunction = info.DrawFunction ~= nil and info.DrawFunction or (info.writeZ == false and DebugLine or DrawLine)
+		end
+
+		for v = 0, 360, 360 / sides do
+			points[iteration] = TransformToParentPoint(transform, Vec(sin(v * degreeToRadian + rotation) * radius, 0, cos(v * degreeToRadian + rotation) * radius))
+			points[iteration + 1] = TransformToParentPoint(transform, Vec(sin(v * degreeToRadian + rotation) * radius, -10, cos(v * degreeToRadian + rotation) * radius))
+			if iteration > 2 then
+				DrawFunction(points[iteration], points[iteration + 1], r, g, b, a)
+				DrawFunction(points[iteration - 2], points[iteration], r, g, b, a)
+				DrawFunction(points[iteration - 1], points[iteration + 1], r, g, b, a)
+			end
+			iteration = iteration + 2
+		end
 	end
 
-	function visual.drawbox(transform, min, max, cr, cg, cb, ca, writeZ)
-		cr = cr or 1
-		cg = cg or 1
-		cb = cb or 1
-		ca = ca or 1
-		local a, b, c, d, e, f, g, h =
-			TransformToParentPoint(transform, Vec(min[1], min[2], min[3])),
-			TransformToParentPoint(transform, Vec(max[1], min[2], min[3])),
-			TransformToParentPoint(transform, Vec(min[1], max[2], min[3])),
-			TransformToParentPoint(transform, Vec(max[1], max[2], min[3])),
-			TransformToParentPoint(transform, Vec(min[1], min[2], max[3])),
-			TransformToParentPoint(transform, Vec(max[1], min[2], max[3])),
-			TransformToParentPoint(transform, Vec(min[1], max[2], max[3])),
-			TransformToParentPoint(transform, Vec(max[1], max[2], max[3]))
-		local DrawFunc = writeZ and DrawLine or DebugLine
-		DrawFunc(a, b, cr, cg, cb, ca)
-		DrawFunc(a, c, cr, cg, cb, ca)
-		DrawFunc(a, e, cr, cg, cb, ca)
-		DrawFunc(d, c, cr, cg, cb, ca)
-		DrawFunc(d, b, cr, cg, cb, ca)
-		DrawFunc(d, h, cr, cg, cb, ca)
-		DrawFunc(f, e, cr, cg, cb, ca)
-		DrawFunc(f, h, cr, cg, cb, ca)
-		DrawFunc(f, b, cr, cg, cb, ca)
-		DrawFunc(g, h, cr, cg, cb, ca)
-		DrawFunc(g, e, cr, cg, cb, ca)
-		DrawFunc(g, c, cr, cg, cb, ca)
+	function visual.drawsphere(transform, quat, radius, rotation, samples, info)
```

Comment:
unlike `visual.drawaxis`, having `quat` here is a problem since it is followed by non-optional parameters. It would probably be best to just keep `transform` and discard the option to pass a vector and a quaternion separately.